src/phoenix/pointcloud/projectors.py

```diff
@@ -1,10 +1,11 @@
+import warnings
 from dataclasses import asdict, dataclass
 from typing import cast
 
 import numpy as np
 import numpy.typing as npt
+from numba.core.errors import NumbaDeprecationWarning, NumbaPendingDeprecationWarning
 from typing_extensions import TypeAlias
-from umap import UMAP
 
```

Comment:
Let's put the `import` inside a context manager at the top level as follows. (Putting the `import` at the top level means it only needs to be called once.) It also suffices to simply ignore all `NumbaWarning` since this is a third party library and we don't really need to mind the specifics.
```suggestion
with warnings.catch_warnings():
    from numba.core.errors import NumbaWarning

    warnings.simplefilter("ignore", category=NumbaWarning)
    from umap import UMAP
```